merlin-server/sql/merlin/tables/constraint.sql

```diff
@@ -82,3 +82,28 @@ create trigger set_timestamp
 before update on "constraint"
 for each row
 execute function constraint_set_updated_at();
+
+create function constraint_check_constraint_run()
+returns trigger
+security definer
+language plpgsql as $$begin
+  if new.definition != old.definition then
+    update constraint_run
+    set definition_outdated = true
+    where constraint_id = new.id
+      and constraint_definition != new.definition
+      and definition_outdated = false;
+  else
+    update constraint_run
+    set definition_outdated = false
+    where constraint_id = new.id
+      and constraint_definition == new.definition
+      and definition_outdated = true;
+  end if;
+  return new;
+end$$;
+
+create trigger constraint_check_constraint_run_trigger
+  after update on "constraint"
+  for each row
```

Comment:
The `when (new.definition != old.definition)` condition I mentioned on the original trigger will still be useful here.